qiskit_experiments/test/mock_iq_backend.py

```diff
@@ -31,7 +33,46 @@
 )
 
 
-class MockRestlessBackend(FakeOpenPulse2Q):
+class FakeOpenPulse2QV2(FakeBackendV2):
+    """BackendV2 version of FakeOpenPulse2Q"""
+
+    def __init__(self):
+        # FakeOpenPulse2Q has its data hard-coded rather than in json files. We
+        # prepopulate the dicts so that FakeBackendV2 does not try to load them
+        # from files.
+        #
+        # We have to use a hack to populate _conf_dict
+        backend_v1 = FakeOpenPulse2Q()
+        self._conf_dict_real = backend_v1._configuration.to_dict()
+        super().__init__()
+        self._props_dict = backend_v1._properties.to_dict()
+        self._defs_dict = backend_v1._defaults.to_dict()
+        self._defaults = backend_v1._defaults
+
+        # Workaround a bug in FakeOpenPulse2Q. It defines u1 on qubit 1 in the
+        # cmd_def in the defaults json file but not in the gates in the
+        # properties json. The code FakeBackendV2 uses to build the Target
+        # assumes these two are consistent.
+        u1_0 = next(g for g in self._props_dict["gates"] if g["gate"] == "u1")
+        self._props_dict["gates"].append(u1_0.copy())
+        self._props_dict["gates"][-1]["qubits"] = [1]
+
+    @property
+    def _conf_dict(self):
+        # FakeBackendV2 sets this in __init__. As a hack, we use this property
+        # to prevent it from overriding our values.
+        return self._conf_dict_real
+
+    @_conf_dict.setter
+    def _conf_dict(self, value):
+        pass
+
+    def defaults(self):
```

Comment:
Since this method is not provided by the provider, I'm not sure if this is really right approach. Actually this mixin generates the mixture of V1 and V2 model, which will never exist outside the QE -- so I'm not sure what is tested by this.

Reply:
`defaults()` is not in the `BackendV1` or `BackendV2` base classes but it is the `BackendV1` and `BackendV2` subclasses produced by `qiskit-ibmq-provider` and `qiskit-ibm-provider`. The main thing that changed is that fake `BackendV1` classes in terra implement `defaults()` while the `BackendV2` versions do not, so switching the tests to the V2 versions required addressing the `defaults()` issue. The classes hold all the `PulseDefaults` data. They just don't expose it through `defaults()`. This is why there is so little code in `fake_pulse_backends.py`.

I think we could probably make a PR to terra to add `FakeOpenPulse2QV2` directly and remove the need for the hacks here (other than adding `defaults()` back like `fake_pulse_backends.py`).

Reply:
> I think we could probably make a PR to terra to add FakeOpenPulse2QV2 directly and remove the need for the hacks here (other than adding defaults() back like fake_pulse_backends.py).

Yes, I think updating terra code (probably provider too) is proper direction. Otherwise, we cannot get `defaults` data set once we switch to actual V2 backends.

Reply:
Oh sorry I didn't notice IBM V2 backend reports defaults object. So what you are doing seems right. We just need to update FakeV2 backends based on IBM one in terra.

https://github.com/Qiskit/qiskit-ibm-provider/blob/06490ab5ccfabc1ceae79bc3e8ba3f8bb82296fa/qiskit_ibm_provider/ibm_backend.py#L714-L735